src/reducers/existingThoughtChange.js

```diff
@@ -51,10 +51,10 @@ export default (state, { oldValue, newValue, context, showContexts, thoughtsRank
     ? contextOf(contextOf(thoughtsRanked)).concat({ value: oldValue, rank: headRank(contextOf(thoughtsRanked)) }).concat(head(thoughtsRanked))
     : contextOf(thoughtsRanked).concat({ value: oldValue, rank })
 
-  const cursorNew = state.cursor && state.cursor.map(thought => thought.value === oldValue && thought.rank === rankInContext
-    ? { value: newValue, rank: thought.rank }
-    : thought
-  )
+  const currentFocusedThought = state.cursor ? state.cursor[state.cursor.length - 1] : null
```

Comment:
@shresthabijay For future reference, we should not use array indexing or slicing with `Path` types. That creates a tight coupling between the implementation of the `Path` and the consumer. Instead, use the accessor methods, `head` and `contextOf`.